drop-transfer/src/service.rs

```diff
@@ -37,15 +36,13 @@ pub(super) struct State {
     pub fdresolv: Option<Arc<crate::file::FdResolver>>,
 }
 
-type TaskRegistry = HashMap<uuid::Uuid, tokio::task::JoinHandle<()>>;
-
 pub struct Service {
     pub(super) state: Arc<State>,
     stop: CancellationToken,
     waiter: AliveWaiter,
     pub(super) logger: Logger,
 
-    tasks: TaskRegistry,
+    refresh_trigger: tokio::sync::watch::Sender<()>,
```

Comment:
Since the three things, namely `trigger`, `stop`, and `waiter` are almost always passed together, maybe it's worth wrapping them into a structure. Thus simplifying the arguments of many functions, like `ws::server::spawn`

Reply:
Oh, sorry the `Sender` cannot be cloned, so there can probably be a method producing the structure consisting of `stop`, `waiter`, and `trigger: Receiver`

Reply:
That might be a worthwhile additional idea to revisit once this MR lands